x-pack/legacy/plugins/code/public/components/codeblock/codeblock.tsx

```diff
@@ -28,33 +28,10 @@ export class CodeBlock extends React.PureComponent<Props> {
   private resizeChecker?: ResizeChecker;
   private currentHighlightDecorations: string[] = [];
 
-  public componentDidMount(): void {
+  public async componentDidMount() {
     if (this.el) {
-      this.ed = monaco.editor.create(this.el!, {
-        value: this.props.code,
-        language: this.props.language,
-        lineNumbers: this.lineNumbersFunc.bind(this),
-        readOnly: true,
-        folding: this.props.folding,
-        minimap: {
-          enabled: false,
-        },
-        scrollbar: {
-          vertical: 'hidden',
-          handleMouseWheel: false,
-          verticalScrollbarSize: 0,
-        },
-        hover: {
-          enabled: false, // disable default hover;
-        },
-        contextmenu: false,
-        selectOnLineNumbers: false,
-        selectionHighlight: false,
-        renderLineHighlight: 'none',
-        renderIndentGuides: false,
-        automaticLayout: false,
-      });
-      this.ed.onMouseDown((e: editor.IEditorMouseEvent) => {
+      await this.tryLoadFile(this.props.code, this.props.language || 'text');
+      this.ed!.onMouseDown((e: editor.IEditorMouseEvent) => {
```

Comment:
do you think we should just `return` if `this.ed` does not exist, instead of assert a couple of `this.ed!` here? the only case when `this.ed` is null is when the  `loadFile` failed twice even with pure text. If so, it's probably ok to do nothing but just return here.

```
    try {
      await monaco.editor.colorize(code, language, {});
      this.loadFile(code, language);
    } catch (e) {
      this.loadFile(code);
    }
```

Reply:
I agree, but one concern is if we return early here, will it be harder to debug if bad things really happened in loadFile?

Reply:
Or throw some errors. Without checking if `this.ed` is undefined or not, it's much easier to crash the app.